### ISSUE TITLE
Allow one time or temporary unblock of sites on the client

### ISSUE THREAD

**Issue by danyal** [AdguardTeam/AdGuardHome#1521]
**Problem**
Sometimes you visit a site and they use a referral or affiliate link which is blocked. It's a hassle to have to manually whitelist the link and if you're not an admin, you're stuck.

**Solution**
Similar to what ublock origin does, allow the user to decide if they want to proceed "temporary" or "whitelist". Ideally temporary allows you to proceed one time but could also be time based.

**Comment by ArchiveBase** [comment]
Solution for it https://github.com/AdguardTeam/AdGuardHome/issues/997 and https://github.com/AdguardTeam/AdGuardHome/issues/435 

> but could also be time based. https://github.com/AdguardTeam/AdGuardHome/issues/1203

You may able to get benefit from https://github.com/AdguardTeam/AdGuardHome/issues/1081 temporally.

**Comment by hoshsadiq** [comment]
Would be great to have it similar to #1879

**Comment by temporallyaccount** [comment]
> Solution for it #997 and #435 but could also be time based. #1203 You may able to get benefit from #1081 temporally.

This would be great solution with pre-configured lists for temporally block or unblock sites and services for particular time.

>Would be great to have it similar to #1879

It'll disable whole filtering feature for selected time period (pi-hole feature as mentioned in it).

I add explanation comment about this feature. first #997 more essential for other DNS server users to gain or provide limited access to web interface than expose access to public or does something complicated with firewalls.
Now AGH beta has block/unblock, block/unblock for this client only options under each row drop down in query log.

:watch:  **Allow one time or temporary unblock of sites on the client** :watch: 
 -  Block/Unblock
 -  Block/Unblock for this client only
 - Block/Unblock for [time] minutes :new: 
 - Block/Unblock [time] min for this client :new: 

Default [time] config options must be in other area in UI for set like we can configure cache size. These block/unblock rules shouldn't write to custom filtering rules.

That's all. i only wanted to explain the idea (not a vote for this feature or i don't have any dislike)

**Comment by temporallyaccount** [comment]
>Solution
Similar to what ublock origin does, allow the user to decide if they want to proceed "temporary" or "whitelist". Ideally temporary allows you to proceed one time but could also be time based.

I see, this feature partially implemented already except time based option.

**Comment by Aikatsui** [comment]
Duplicate of #2823

**Comment by hoshsadiq** [comment]
@Aikatsui that issue is different. That's adding an individual domain to the allow list temporarily (or disabling a rule temporarily), whereas this issue is globally disabling AGH temporarily.

**Comment by hoshsadiq** [comment]
Heh, I stand corrected. I thought it was a different issue.

**Comment by ainar-g** [comment]
That other one seems to already have a milestone attached, so we'll probably merge this one into that one.  Thanks, everyone.
